### ISSUE TITLE
Problem with thermostat page

### ISSUE THREAD

**Issue by pvtex** [Blackymas/NSPanel_HA_Blueprint#2167]
### TFT Version

4.3.9

### ESPHome Version

4.3.9

### Blueprint Version

4.3.9

### Panel Model

EU

### What is the bug?

when i open the thermostat page for the main climate enity of the nspanel i only see a a horizonal slider (from blue to red with a minus sign left and a plus sign right).
no temperatures are shown and i can not change the temperature.

i  noticed that the build parameter (target_temp_tep for example) are not used. in the ha developer tools is the target_temp_step still 0.1

### Steps to Reproduce

flash the nspanel and use the attached yaml.

### Your Panel's YAML

```yaml
esp32:
  framework:
    type: esp-idf

substitutions:
  ###### CHANGE ME START ######
  device_name: nspanel
  friendly_name: NSPANEL
  name: nspanel
  wifi_ssid: !secret wifi_ssid
  wifi_password: !secret wifi_password
  ota_password: "REMOVED"

  nextion_update_url: "http://homeassistant.local:8123/local/nspanel_eu.tft"
  nextion_blank_url: "http://homeassistant.local:8123/local/nspanel_blank.tft"

  # Add-on configuration (if needed)
  ## Add-on climate
  heater_relay: "2"     # Possible values: "1" or "2"
  temp_units: "°C"      # Temperatures in Fahrenheit
  temp_min: "10"        # Min supported temperature is 40°F
  temp_max: "30"        # Max supported temperature is 80°F
  temp_step: "1"        # Temperature granularity is 1°F
  heat_deadband: "1"  # Temperature delta before engaging heat
  heat_overrun: "1"   # Temperature delta before disengaging heat

# Customization area
##### My customization - Start #####
api:
  encryption:
    key: "REMOVED"

ota:
  - platform: esphome
    password: "8f786bb1017396bd426c9bb285753a5c"
##### My customization - End #####

climate:
  - id: !extend thermostat_embedded
    visual:
      temperature_step: ${temp_step}
    preset:
      - name: Home
        default_target_temperature_low: 18
        mode: "heat"
      - name: Away
        default_target_temperature_low: 16
        mode: "heat"

# Core and optional configurations
packages:
  remote_package:
    url: https://github.com/Blackymas/NSPanel_HA_Blueprint
    ref: main
    refresh: 300s
    files:
      - nspanel_esphome.yaml # Basic package
      # Optional advanced and add-on configurations
      # - esphome/nspanel_esphome_advanced.yaml
      # - esphome/nspanel_esphome_addon_ble_tracker.yaml
      # - esphome/nspanel_esphome_addon_bluetooth_proxy.yaml
      # - esphome/nspanel_esphome_addon_climate_cool.yaml
      - esphome/nspanel_esphome_addon_climate_heat.yaml
      # - esphome/nspanel_esphome_addon_climate_dual.yaml
      # - esphome/nspanel_esphome_addon_cover.yaml
```

### ESPHome Logs

2024-06-21 07:25:40,006 INFO 200 GET /devices (0.0.0.0) 2.57ms
2024-06-21 07:30:40,005 INFO 200 GET /devices (0.0.0.0) 2.10ms
2024-06-21 07:35:40,006 INFO 200 GET /devices (0.0.0.0) 2.50ms
2024-06-21 07:40:40,005 INFO 200 GET /devices (0.0.0.0) 1.86ms
2024-06-21 07:45:40,004 INFO 200 GET /devices (0.0.0.0) 1.72ms
2024-06-21 07:50:40,007 INFO 200 GET /devices (0.0.0.0) 2.89ms
2024-06-21 07:55:40,006 INFO 200 GET /devices (0.0.0.0) 2.22ms
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service base-addon-banner: starting

-----------------------------------------------------------
 Add-on: ESPHome
 ESPHome add-on for intelligently managing all your ESP8266/ESP32 devices
-----------------------------------------------------------
 Add-on version: 2024.6.1
 You are running the latest version of this add-on.
 System: Home Assistant OS 12.4  (amd64 / qemux86-64)
 Home Assistant Core: 2024.6.3
 Home Assistant Supervisor: 2024.06.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
s6-rc: info: service base-addon-banner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service base-addon-timezone: starting
s6-rc: info: service base-addon-log-level: starting
s6-rc: info: service fix-attrs successfully started
[10:02:55] INFO: Configuring timezone (Europe/Berlin)...
s6-rc: info: service base-addon-log-level successfully started
s6-rc: info: service base-addon-timezone successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/30-esphome-fork.sh
cont-init: info: /etc/cont-init.d/30-esphome-fork.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-nginx: starting
s6-rc: info: service esphome: starting
s6-rc: info: service esphome successfully started
[10:02:58] INFO: Migrating old .esphome directory...
s6-rc: info: service init-nginx successfully started
s6-rc: info: service nginx: starting
s6-rc: info: service nginx successfully started
s6-rc: info: service discovery: starting
mv: cannot stat '/config/esphome/.esphome/*.json': No such file or directory
[10:02:58] INFO: Waiting for ESPHome dashboard to come up...
[10:02:58] INFO: Starting ESPHome dashboard...
2024-06-21 10:03:01,785 INFO Starting dashboard web server on unix socket /var/run/esphome.sock and configuration dir /config/esphome...
[10:03:01] INFO: Starting NGINX...
[10:03:02] INFO: Successfully send discovery information to Home Assistant.
s6-rc: info: service discovery successfully started
s6-rc: info: service legacy-services: starting
s6-rc: info: service legacy-services successfully started
2024-06-21 10:04:03,191 INFO 200 GET /devices (0.0.0.0) 1.90ms
2024-06-21 10:09:05,028 INFO 200 GET /devices (0.0.0.0) 1.81ms
2024-06-21 10:14:05,027 INFO 200 GET /devices (0.0.0.0) 1.57ms
2024-06-21 10:19:05,031 INFO 200 GET /devices (0.0.0.0) 2.37ms
2024-06-21 10:24:05,027 INFO 200 GET /devices (0.0.0.0) 1.68ms
2024-06-21 10:29:05,027 INFO 200 GET /devices (0.0.0.0) 2.11ms
2024-06-21 10:34:05,026 INFO 200 GET /devices (0.0.0.0) 1.70ms
2024-06-21 10:39:05,026 INFO 200 GET /devices (0.0.0.0) 1.59ms
2024-06-21 10:44:05,027 INFO 200 GET /devices (0.0.0.0) 1.82ms
2024-06-21 10:49:05,028 INFO 200 GET /devices (0.0.0.0) 1.88ms
2024-06-21 10:51:12,451 INFO 304 GET / (0.0.0.0) 26.94ms
2024-06-21 10:51:12,574 INFO 200 GET /devices (0.0.0.0) 13.23ms
2024-06-21 10:51:12,622 INFO 101 GET /compile (0.0.0.0) 0.77ms
2024-06-21 10:51:12,743 INFO Running command 'esphome --dashboard compile --only-generate /config/esphome/nspanel_esphome.yaml'
2024-06-21 10:51:17,566 INFO 304 GET /devices (0.0.0.0) 2.40ms
2024-06-21 10:51:20,227 INFO Process exited with return code 0
2024-06-21 10:51:20,251 INFO 101 GET /compile (0.0.0.0) 1.01ms
2024-06-21 10:51:20,254 INFO Running command 'esphome --dashboard compile --only-generate /config/esphome/nspanel_esphome.yaml'
2024-06-21 10:51:22,281 INFO 200 GET /edit?configuration=nspanel.yaml (0.0.0.0) 4.33ms
2024-06-21 10:51:22,338 INFO 101 GET /ace (0.0.0.0) 0.80ms
2024-06-21 10:51:22,347 INFO Running command 'esphome --dashboard -q vscode --ace /config/esphome'
2024-06-21 10:51:22,841 INFO Process exited with return code 0
2024-06-21 10:52:27,943 INFO 200 GET /devices (0.0.0.0) 7.52ms
2024-06-21 10:52:32,912 INFO 304 GET /devices (0.0.0.0) 2.22ms
2024-06-21 10:52:37,920 INFO 304 GET /devices (0.0.0.0) 1.40ms
2024-06-21 10:52:43,706 INFO 304 GET /devices (0.0.0.0) 1.76ms
2024-06-21 10:52:49,698 INFO 304 GET /devices (0.0.0.0) 5.17ms
2024-06-21 10:52:55,709 INFO 304 GET /devices (0.0.0.0) 3.78ms
2024-06-21 10:53:01,706 INFO 304 GET /devices (0.0.0.0) 1.59ms
2024-06-21 10:53:07,714 INFO 304 GET /devices (0.0.0.0) 10.48ms
2024-06-21 10:53:13,709 INFO 304 GET /devices (0.0.0.0) 1.69ms
2024-06-21 10:53:19,695 INFO 304 GET /devices (0.0.0.0) 1.39ms
2024-06-21 10:53:25,699 INFO 304 GET /devices (0.0.0.0) 1.70ms
2024-06-21 10:53:31,698 INFO 304 GET /devices (0.0.0.0) 2.34ms
2024-06-21 10:53:37,701 INFO 304 GET /devices (0.0.0.0) 3.59ms
2024-06-21 10:53:43,701 INFO 304 GET /devices (0.0.0.0) 2.01ms
2024-06-21 10:53:49,708 INFO 304 GET /devices (0.0.0.0) 2.62ms
2024-06-21 10:53:55,701 INFO 304 GET /devices (0.0.0.0) 1.95ms
2024-06-21 10:54:01,705 INFO 304 GET /devices (0.0.0.0) 2.04ms
2024-06-21 10:54:05,027 INFO 200 GET /devices (0.0.0.0) 1.58ms
2024-06-21 10:54:07,708 INFO 304 GET /devices (0.0.0.0) 1.79ms
2024-06-21 10:54:13,698 INFO 304 GET /devices (0.0.0.0) 1.33ms
2024-06-21 10:54:19,697 INFO 304 GET /devices (0.0.0.0) 2.06ms
2024-06-21 10:54:25,705 INFO 304 GET /devices (0.0.0.0) 1.55ms
2024-06-21 10:54:30,710 INFO 304 GET /devices (0.0.0.0) 1.68ms
2024-06-21 10:54:32,345 INFO 304 GET /edit?configuration=nspanel.yaml (0.0.0.0) 1.69ms
2024-06-21 10:54:32,380 INFO 101 GET /ace (0.0.0.0) 0.44ms
2024-06-21 10:54:32,384 INFO Running command 'esphome --dashboard -q vscode --ace /config/esphome'

**Comment by pvtex** [comment]
found the error.
my climate entity had an error

**Comment by edwardtfn** [comment]
There is an issue with ESPHome preventing the extend when a list is used (like you have with preset). That was already fixed in dev, so I hope will be released with ESPHome v2024.6.2.
That give a different error message, but could be the case here.

- esphome/issues#5907
